src/mon/OSDMonitor.cc

```diff
@@ -6660,7 +6660,13 @@ int OSDMonitor::prepare_pool_size(const unsigned pool_type,
       err = get_erasure_code(erasure_code_profile, &erasure_code, ss);
       if (err == 0) {
 	*size = erasure_code->get_chunk_count();
-	*min_size = std::min(erasure_code->get_data_chunk_count() + 1, *size);
+	*min_size =
+	  erasure_code->get_data_chunk_count() +
+	  std::max<int>(
+	    0,
+	    std::min<int>(1, erasure_code->get_coding_chunk_count() - 1));
```

Comment:
i am surprised to find that we can set m=0 when creating an erasure profile! and m=-1 is also accepted!

http://tracker.ceph.com/issues/38682 is created for tracking this issue.

Reply:
if m is always greater than 0. we can safely put
```c++
*min_size = (erasure_code->get_data_chunk_count() +
             std::min(1u, erasure_code->get_coding_chunk_count() - 1)));
```